bitcoin/script/miniscript.h

```diff
@@ -765,7 +765,10 @@ struct Node {
     bool NeedsSignature() const { return GetType() << "s"_mst; }
 
     //! Do all sanity checks.
-    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize(); }
+    bool IsSafe() const { return GetType() << "ms"_mst && CheckOpsLimit() && CheckStackSize(); }
```

Comment:
Perhaps, to make things more in line with the terminology in the Rust implementation, call this function `IsSane` and `IsSaneTopLevel` ?

Also, perhaps include `IsValid() &&`? Or is that not necessary?

Reply:
Actually, I think `IsSane()` should not include the "s" propery requirement - most scripts will contain nodes that are non-"s" (but the top level will be, so more it to `IsSaneTopLevel()`; "m" can stay in `IsSane()`).

Reply:
Ack renaming

I thought that including `IsValid()` was redundant (if we add a check for validity in #68 too) but you can still create a `Node` and really any `IsSane` script should be `IsValid`, so i think it's good to have it there.

Right.

Reply:
Done